rust/linux-client/src/main.rs

```diff
@@ -49,13 +50,33 @@ async fn main() -> Result<()> {
     )
     .unwrap();
 
-    tokio::signal::ctrl_c().await?;
+    let mut sigint = tokio::signal::unix::signal(SignalKind::interrupt())?;
```

Comment:
I think we should do this kind of examples in `example` binaries rather than adding more things to the binary.

Reply:
Or is there a case where we want to do a restart with a signal?

Reply:
I think `systemctl reload` sends a SIGHUP to reload config without stopping the process? Can't find docs for it though.

Reply:
> Or is there a case where we want to do a restart with a signal?

My research concluded that it is pretty common for daemons to accept SIGHUP for reloading config. I don't know about systemd doing it but it didn't feel too wrong adding it :)